Sanitize negative followers count on APContact

realkinetix opened the pull request:
Please see
https://github.com/friendica/friendica/issues/9498#issuecomment-818894106
and related discussion regarding this - it appears it's possible for AP
users, maybe just Mastodon users, to have a negative followers count.
This causes fatal errors in Friendica, so I think we should sanitize
this input.

realkinetix committed:
Sanitize negative followers count on APContact

Please see
https://github.com/friendica/friendica/issues/9498#issuecomment-818894106
and related discussion regarding this - it appears it's possible for AP
users, maybe just Mastodon users, to have a negative followers count.
This causes fatal errors in Friendica, so I think we should sanitize
this input.

MrPetovan commented on src/Model/APContact.php:
```diff
@@ -272,6 +272,11 @@ public static function getByURL($url, $update = null)
+			// Mastodon seriously allows for this condition? 
+			// Jul 14 2021 - See https://mastodon.online/@goes11 for a negative followers count
+			if ($followers['totalItems'] < 0) {
+			  $followers['totalItems'] = 0;
```
Code standards: please use tabulations for indentation.
```suggestion
				$followers['totalItems'] = 0;
```

realkinetix replied:
Shoot, sorry about that!

MrPetovan replied:
It's okay, that's why code review exists!

realkinetix committed:
Fix indent to standards

realkinetix committed:
Adjust where the code is to avoid empty warning

Shifted the code to avoid:
PHP Warning:  Undefined array key "totalItems" in /home/friendicadev/friendica/src/Model/APContact.php on line 277

realkinetix committed:
Negative numbers in following is a thing too

Check & sanitize 'following' as well, as per:
https://mastodon.social/@BLUW

MrPetovan commented:
Thank you for your work!
